content/en/blog/2023/how_to_debug_large_db_size_issue.md

```diff
@@ -134,6 +141,9 @@ Notes:
 1. Compaction is a cluster-wide operation, so you only need to execute compaction once on whichever etcd member.
 Of course, it will not do any harm if executing it multiple times.
 2. Defragmentation is a time-consuming task, so it's recommended to do it for each member one by one.
+3. Defragmentation is an expensive operation, so you should do it as infrequent as possible. On the other hand,
+you also need to make sure any etcd member will not run out of the storage quota. It's recommended to run
+defragmentation using a tool such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag/).
```

Comment:
@ahrtr Love the tool. May be some day this be available in the etcd repo :) Not sure if you already have thoughts around it. For this doc, I noticed that doc link for `etcd defrag` is already provided so that's good. But, a nit, I think may be a good idea to say something like "etcdutl defrag can be used for defragmentation, but tools such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag/) can provide enhanced functionality and ease of use and should be considered to use.  (vs recommended considering I am not sure how many people have already used it and if we have feedback) I am good either way but just a thought. I will let you decide. Thanks!

Reply:
Thanks @spzala for the quick review.

> May be some day this be available in the etcd repo :) Not sure if you already have thoughts around it.

Yes, I thought about it, but do not figure out what's the best place to put or integrate the tool. So creating it as a separate tool/repo is the best choice as for now I can think of.

> I think may be a good idea to say something like "etcdutl defrag can be used for defragmentation, but tools such as [etcd-defrag](https://github.com/ahrtr/etcd-defrag/) can provide enhanced functionality and ease of use and should be considered to use.

Makes sense & updated.

> I am not sure how many people have already used it and if we have feedback.

defragmentation is a hot topic in etcd community, such as https://github.com/etcd-io/etcd/issues/15477. There are lots of problematic solutions (basically simple wraper of the etcdctl command) in github, so eventually I decided to create the tool myself. It's just about 7 days old:) , but got couple of positive feedback.

Reply:
That's cool and great work @ahrtr , thank you!!! And, changes LGTM.